internal/tiltfile/files.go

```diff
@@ -401,6 +402,40 @@ func (s *tiltfileState) listdir(thread *starlark.Thread, fn *starlark.Builtin, a
 	return starlark.NewList(ret), nil
 }
 
+func (s *tiltfileState) readYaml(thread *starlark.Thread, fn *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var path starlark.String
+	var defaultValue starlark.Value
+	if err := starlark.UnpackArgs(fn.Name(), args, kwargs, "path", &path, "default?", &defaultValue); err != nil {
+		return nil, err
+	}
+
+	localPath, err := s.localPathFromSkylarkValue(path)
+	if err != nil {
+		return nil, fmt.Errorf("Argument 0 (path): %v", err)
+	}
+
+	contents, err := s.readFile(localPath)
+	if err != nil {
+		// Return the default value if the file doesn't exist AND a default value was given
+		if os.IsNotExist(err) && defaultValue != nil {
+			return defaultValue, nil
+		}
+		return nil, err
+	}
+
+	var decodedJSON interface{}
+	err = yaml.Unmarshal(contents, &decodedJSON)
+	if err != nil {
+		return nil, fmt.Errorf("YAML to JSON conversion error: %v in %s", err, path.GoString())
```

Comment:
A user might be confused that reading YAML involves JSON. Should we hide that from them and just call this "error parsing YAML"? I'm not sure.

Reply:
Yeah, I think that makes sense. I'll change it

Reply:
Later on in the function there's another error messages that says "error converting JSON to Starlark..." and it might be useful if I change that one as well?

Reply:
Yeah sounds good to me.